src/System.Linq/tests/ToListTests.cs

```diff
@@ -25,6 +25,16 @@ public void ToList_AlwaysCreateACopy()
         }
 
 
+        [Fact]
+        public void ToList_Capacity()
+        {
+            List<int> sourceList = new List<int>() { 1, 2, 3, 4, 5 };
+            List<int> resultList = sourceList.ToList(20);
+
+            Assert.Equal(20, resultList.Capacity);
+            Assert.Equal(sourceList, resultList);
```

Comment:
Ideally there should be tests with the source types that have different `ToList()` behaviours, tests asking for a smaller capacity than that sought, and error conditions.